src/main/java/io/cdap/plugin/successfactors/source/SuccessFactorsSource.java

```diff
@@ -136,7 +136,16 @@ private Schema getOutputSchema(FailureCollector failureCollector) {
       } catch (TransportException te) {
         String errorMsg = ExceptionParser.buildTransportError(te);
         errorMsg = ResourceConstants.ERR_ODATA_SERVICE_CALL.getMsgForKeyWithCode(errorMsg);
-        failureCollector.addFailure(errorMsg, null).withConfigProperty(SuccessFactorsPluginConfig.BASE_URL);
+        if ((errorMsg.contains("proxy") || errorMsg.contains("Connection reset"))) {
```

Comment:
Is there a documentation that maps these KEY words to the given properties ? 

`Connection reset` , `Failed to connect` can be very generic error messages and may not be related to the configs...

Reply:
As tested the errors thrown in these scenarios are always same and to highlight the errors on the particular fields these keywords needs to be used.

Reply:
That might be true, 
but these key words may occur for other failures NOT related to proxy. 
Like network failure or something.

In such cases, these suggestion might mislead a user in debugging in wrong direction.

Reply:
If there is no pre-defined error message for proxy related stuff, then Please surface the whole error. 
It's not correct to assume error message, as these can change in future unless it's mentioned by their official doc.